### PR TITLE
chore(deps): upgrade to symfony 5 and fosrestbundle 3

### PR DESCRIPTION
**Pull request by kduret**
## Description

upgrade to symfony 5 and fosrestbundle 3

**Fixes** MON-6124

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Check CI

### CODE REVIEW

File: src/Centreon/Domain/Contact/Contact.php [hunk 1]
```diff
@@ -650,4 +650,12 @@ public function setOneClickExportEnabled(bool $isOneClickExportEnabled): static
 
         return $this;
     }
+
+    /**
+     * {@inheritDoc}
```
Comment (kduret) [inline src/Centreon/Domain/Contact/Contact.php:655]:
```suggestion
     * @inheritDoc
```

File: src/Security/SessionAPIAuthenticator.php [hunk 2]
```diff
@@ -78,111 +81,85 @@ public function __construct(
...
+        if (!$this->authenticationService->isValidToken($sessionId)) {
+            throw new BadCredentialsException();
         }
 
         $this->sessionRepository->deleteExpiredSession();
 
         $contact = $this->contactRepository->findBySession($sessionId);
         if ($contact === null) {
-            throw new SessionUnavailableException();
+            throw new UserNotFoundException();
```
Comment (callapa) [inline src/Security/SessionAPIAuthenticator.php:160]: It is not the user that is not found but the session.

File: src/Security/SessionAPIAuthenticator.php [hunk 3]
```diff
@@ -78,111 +81,85 @@ public function __construct(
...
         }
-        if ($contact->isActive() === false) {
+        if (!$contact->isActive()) {
```
Comment (callapa) [inline src/Security/SessionAPIAuthenticator.php:162]: === false is more readable

File: src/Security/TokenAPIAuthenticator.php [hunk 4]
```diff
@@ -57,237 +59,121 @@ class TokenAPIAuthenticator extends AbstractGuardAuthenticator
...
      *
      * @param AuthenticationRepositoryInterface $authenticationRepository
      * @param ContactRepositoryInterface $contactRepository
-     * @param OptionServiceInterface $optionService
      */
     public function __construct(
         AuthenticationRepositoryInterface $authenticationRepository,
         ContactRepositoryInterface $contactRepository,
-        OptionServiceInterface $optionService
+        LocalProvider $localProvider
```
Comment (callapa) [inline src/Security/TokenAPIAuthenticator.php:75]: please complete PHPDoc

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Security/SessionAPIAuthenticator.php (hunk 2 above)
Comment (callapa) [reply src/Security/SessionAPIAuthenticator.php:160]: You can separate, first try to find the session and then the user

**Review by adr-mo — DISMISSED** [review]

**Review by AssanDiomande — DISMISSED** [review]

File: src/Centreon/Domain/Authentication/UseCase/Authenticate.php [hunk 5]
```diff
@@ -320,8 +326,9 @@ private function createUserOrFail(ProviderInterface $authenticationProvider, Con
     private function startLegacySession(Centreon $legacySession): void
     {
         $this->info('[AUTHENTICATE] Starting Centreon Session');
-        $this->session->start();
-        $_SESSION['centreon'] = $legacySession;
+        $this->requestStack->getCurrentRequest()->getSession()->start();
+        $this->requestStack->getCurrentRequest()->getSession()->set('centreon', $legacySession);
+        $_SESSION['centreon'] = &$legacySession;
```
Comment (kduret) [inline src/Centreon/Domain/Authentication/UseCase/Authenticate.php:331]:
```suggestion
        $_SESSION['centreon'] = $legacySession;
```

File: config/packages/Centreon.yaml [hunk 6]
```diff
@@ -3,20 +3,20 @@ jms_serializer:
...
-                path: '%kernel.root_dir%/../config/packages/serializer/Infrastructure'
+                path: '%kernel.project_dir%/config/packages/serializer/Infrastructure'
 parameters:
     api.header: "Api-Version"
     api.version.latest: "21.10"
     debug_log_file: '%log_path%/centreon-web.log'
     env(DEBUG_LEVEL): !php/const Monolog\Logger::NOTICE
     curl.timeout: 60
-    debug_level: "%env(DEBUG_LEVEL)%"
+    debug_level: "%env(string:DEBUG_LEVEL)%"
```
Comment (callapa) [inline config/packages/Centreon.yaml:19]:
```suggestion
    debug_level: "%env(int:DEBUG_LEVEL)%"
```

File: src/Centreon/Domain/Authentication/UseCase/Authenticate.php [hunk 7]
```diff
@@ -95,7 +95,7 @@ class Authenticate
      * @param AuthenticationServiceInterface $authenticationService
      * @param ProviderServiceInterface $providerService
      * @param ContactServiceInterface $contactService
-     * @param SessionInterface $session
+     * @param RequestStack $session
```
Comment (callapa) [inline src/Centreon/Domain/Authentication/UseCase/Authenticate.php:98]:
```suggestion
     * @param RequestStack $requestStack
```

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by callapa — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**